config/security/token.go

```diff
@@ -0,0 +1,85 @@
+package security
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"github.com/kiali/kiali/log"
+	"time"
+)
+
+type AuthToken struct {
+	Info string // Contains TokenInfo in base 64 encoded json
+	HMAC string // Base 64 encoded hmac
+}
+
+type TokenInfo struct {
+	UserID         string
+	ExpirationDate time.Time
+}
+
+func NewAuthToken(uid string, expirationDate time.Time, secret string) *AuthToken {
+	at := &AuthToken{
+		Info: NewTokenInfo(uid, expirationDate).ToBase64(),
+	}
+	at.HMAC = ComputeHmac256(at.Info, secret)
+	return at
+}
+
+func NewTokenInfo(uid string, expirationDate time.Time) *TokenInfo {
+	return &TokenInfo{
+		UserID:         uid,
+		ExpirationDate: expirationDate,
+	}
+}
+
+func (at *AuthToken) verify(secret string) bool {
+	if ComputeHmac256(at.Info, secret) == at.HMAC {
+		return true
+	} else {
+		return false
+	}
+}
+
+func (at *AuthToken) GetTokenInfo(secret string) (*TokenInfo, error) {
```

Comment:
Trying to find a flaw, but it seems good :)
However I wonder if we could use an existing JWT-like implementation that would be more widely tested & trusted.

Reply:
keycloak not support golang I didn't find any JWT implementation in golang only some methods of thirds....

Reply:
I'd also prefer an existing JWT implementation.
This page list several ones for golang: https://jwt.io/

Reply:
Mmm there are only libraries of thirds...I am gonna check them